internal/kubernetes/operator/datafederation/datafederation.go

```diff
@@ -0,0 +1,201 @@
+}
+
+func isDataFederationExportable(dataFederation *admin.DataLakeTenant) bool {
+	state := dataFederation.GetState()
+	if state == DeletingState || state == DeletedState {
+		return false
+	}
+	return true
+}
+
+func GetDataFederationSpec(dataFederationSpec *admin.DataLakeTenant, targetNamespace, projectName string) atlasV1.DataFederationSpec {
+	return atlasV1.DataFederationSpec{
+		Project:             common.ResourceRefNamespaced{Name: projectName, Namespace: targetNamespace},
+		Name:                dataFederationSpec.GetName(),
+		CloudProviderConfig: GetCloudProviderConfig(dataFederationSpec.CloudProviderConfig),
+		DataProcessRegion:   GetDataProcessRegion(dataFederationSpec.DataProcessRegion),
+		Storage:             GetStorage(dataFederationSpec.Storage),
+	}
+}
+
+func GetCloudProviderConfig(cloudProviderConfig *admin.DataLakeCloudProviderConfig) *atlasV1.CloudProviderConfig {
+	if cloudProviderConfig == nil {
+		return &atlasV1.CloudProviderConfig{}
+	}
+	return &atlasV1.CloudProviderConfig{
+		AWS: &atlasV1.AWSProviderConfig{
+			RoleID:       cloudProviderConfig.Aws.RoleId,
+			TestS3Bucket: cloudProviderConfig.Aws.TestS3Bucket,
+		},
+	}
+}
+
+func GetDataProcessRegion(dataProcessRegion *admin.DataLakeDataProcessRegion) *atlasV1.DataProcessRegion {
+	if dataProcessRegion == nil {
+		return &atlasV1.DataProcessRegion{}
+	}
+	return &atlasV1.DataProcessRegion{
+		CloudProvider: dataProcessRegion.CloudProvider,
+		Region:        dataProcessRegion.Region,
+	}
+}
+
+func GetStorage(storage *admin.DataLakeStorage) *atlasV1.Storage {
+	if storage == nil {
+		return &atlasV1.Storage{}
+	}
+	return &atlasV1.Storage{
+		Databases: GetDatabases(storage.Databases),
+		Stores:    GetStores(storage.Stores),
+	}
+}
+func GetDatabases(database []admin.DataLakeDatabaseInstance) []atlasV1.Database {
+	if database == nil {
+		return []atlasV1.Database{}
+	}
+	var result []atlasV1.Database
+	for _, obj := range database {
+		result = append(result, atlasV1.Database{
+			Collections:            GetCollection(obj.GetCollections()),
+			MaxWildcardCollections: obj.GetMaxWildcardCollections(),
+			Name:                   obj.GetName(),
+			Views:                  GetViews(obj.GetViews()),
+		})
+	}
+	return result
+}
+
+func GetCollection(collections []admin.DataLakeDatabaseCollection) []atlasV1.Collection {
+	if collections == nil {
+		return []atlasV1.Collection{}
+	}
+	var result []atlasV1.Collection
+	for _, obj := range collections {
+		result = append(result, atlasV1.Collection{
+			DataSources: GetDataSources(obj.GetDataSources()),
+			Name:        obj.GetName(),
+		})
+	}
+	return result
+}
+
+func GetDataSources(dataSources []admin.DataLakeDatabaseDataSourceSettings) []atlasV1.DataSource {
+	if dataSources == nil {
+		return []atlasV1.DataSource{}
+	}
+	var result []atlasV1.DataSource
+	for _, obj := range dataSources {
+		result = append(result, atlasV1.DataSource{
+			AllowInsecure:       aws.ToBool(obj.AllowInsecure),
```

Comment:
save yourself a lib you don't need
```suggestion
			AllowInsecure:       obj.GetAllowInsecure(),
```